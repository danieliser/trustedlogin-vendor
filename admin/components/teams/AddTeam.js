import { useSettings } from "../../hooks/useSettings";
import { useView } from "../../hooks/useView";
import EditTeam from "./EditTeam";

const AddTeam = () => {
  const { addTeam } = useSettings();
  const { setCurrentView } = useView();
  const onClickSave = (newTeam) => {
    addTeam(newTeam, true);
    setCurrentView("teams");
  };
  return <EditTeam onClickSave={onClickSave}
    formTitle={"Add Team"} />;
};

export default AddTeam;
