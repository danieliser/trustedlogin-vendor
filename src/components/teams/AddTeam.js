import { useSettings } from "../../hooks/useSettings";
import { useView } from "../../hooks/useView";
import { SecondaryButton } from "../Buttons";
import EditTeam, { TeamFormRows } from "./EditTeam";
import { useMemo, useState } from "react";
import TitleDescriptionLink from "../TitleDescriptionLink";
import useRemoteSession, {
  ReloadIfNoSessionData,
} from "../../hooks/useRemoteSession";
import LoginOrLogout from "../LoginLogout";
import { InputField } from "./fields";
import teamFields from "./teamFields";
import Connector from "../connect/index";

const CreateTeam = () => {
  const { hasAppToken, session } = useRemoteSession();
  const formRef = useRef(null);
  if (!hasAppToken) {
    return (
      <>
        <LoginOrLogout />
      </>
    );
  }
  const handler = (e) => {
    //Check if form input is valid
    if (!formRef.current.checkValidity()) {
      //If not, return, allowing browser's native validation errors to show
      return;
    }
    //Now, prevent default form submission.
    //Can not do this before checkValidity, because that will prevent the browser's native validation errors from showing.
    e.preventDefault();
    const formData = new FormData(formRef.current);
    const data = Object.fromEntries(formData);
    console.log(data);
  };
  return (
    <>
      <form formRef={formRef} onSubmit={handler}>
        <InputField
          type="text"
          name={teamFields.name.id}
          id={teamFields.name.id}
          label={teamFields.name.label}
          defaultValue={""}
          required={true}
        />
      </form>
    </>
  );
};
const AddTeam = () => {
  const { addTeam } = useSettings();
  const { setCurrentView } = useView();
  //null|create|connect
  const [addType, setAddType] = useState(null);

  const title = useMemo(() => {
    if ("create" === addType) {
      return "Create Team";
    }
    if ("connect" === addType) {
      return "Connect Team";
    }
    return "Are you connecting an existing team to this site or do you need to you need to create a team?";
  }, [addType]);

  const cancel = useMemo(() => {
    if ("create" === addType || "connect" === addType) {
      return {
        label: "Back",
        onClick: (e) => {
          e.preventDefault();
          setAddType(null);
        },
      };
    }
    return {
      label: "Cancel",
      onClick: (e) => {
        e.preventDefault();
        setCurrentView("teams");
      },
    };
  }, [setCurrentView, addType]);
  const onClickConnect = (e) => {
    e.preventDefault();
    setAddType("connect");
  };
  const onClickCreate = (e) => {
    e.preventDefault();
    setAddType("create");
  };

  const Inside = useMemo(() => {
    return () => {
      if ("create" === addType) {
        return (
          <>
            <CreateTeam />
          </>
        );
      }
      if ("connect" === addType) {
        return (
          <>
            <Connector />
          </>
        );
      }
      return (
        <TeamFormRows
          Left={() => (
            <>
              <SecondaryButton onClick={onClickCreate}>
                Create Team
              </SecondaryButton>
            </>
          )}
          Right={() => (
            <>
              <SecondaryButton onClick={onClickConnect}>
                Connect Existing Team
              </SecondaryButton>
            </>
          )}
        />
      );
    };
  }, [addType]);

  return (
    <>
      <ReloadIfNoSessionData />
      <div className="flex px-5 pt-20 sm:px-10">
        <div className="flex flex-col w-full max-w-4xl mx-auto p-8 bg-white rounded-lg shadow sm:p-14 sm:pb-8">
          <svg
            className="mx-auto"
            width="56"
            height="56"
            viewBox="0 0 56 56"
            fill="none"
            xmlns="http://www.w3.org/2000/svg">
            <rect x="4" y="4" width="48" height="48" rx="24" fill="#00AADD" />
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
              stroke="white"
              strokeWidth="2"
            />
            <rect
              x="4"
              y="4"
              width="48"
              height="48"
              rx="24"
              stroke="#CDEFF9"
              strokeWidth="8"
            />
          </svg>
          <TitleDescriptionLink
            title={title}
            link={"https://app.trustedlogin.com/settings#/teams"}
          />
          <Inside />
          <div className="flex flex-col py-6 space-y-6 sm:space-y-0 sm:space-x-12 sm:flex-row">
            <SecondaryButton onClick={cancel.onClick}>
              {cancel.label}
            </SecondaryButton>
          </div>
        </div>
      </div>
    </>
  );
};

export default AddTeam;
