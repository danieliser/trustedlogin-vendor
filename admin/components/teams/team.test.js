import AddTeam from "./AddTeam";
import {render} from '@testing-library/react';
import ViewProvider from "../../hooks/useView";
import SettingsProvider from "../../hooks/useSettings";
import NoTeams from "./NoTeams";

const Provider = ({ children }) => (
    <SettingsProvider>
        <ViewProvider>
            {children}

        </ViewProvider>

    </SettingsProvider>
);
const team ={
account_id: "1",
private_key: "asda",
public_key: "fsaffff",
helpdesk: "HelpScout",
approved_roles: [],
}
describe( 'AddTeam', () => {
    it( 'Renders & Matches snapshot', () => {
        const {container} =render(<Provider>
            <AddTeam />
        </Provider>);
        expect( container ).toMatchSnapshot();
    });
})
