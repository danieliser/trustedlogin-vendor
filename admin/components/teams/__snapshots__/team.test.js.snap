// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTeam Renders & Matches snapshot 1`] = `
<div>
  <form
    class="flex px-5 pt-20 sm:px-10"
  >
    <div
      class="flex flex-col w-full max-w-4xl mx-auto p-8 bg-white rounded-lg shadow sm:p-14 sm:pb-8"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <div
        class="max-w-sm mx-auto mb-8 justify-center text-center"
      >
        <h2
          class="mt-4 text-2xl text-gray-900"
        >
          Add Team
        </h2>
        <p
          class="mt-2 mb-4 text-sm text-gray-500"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ornare tortor in nisl fermentum.
        </p>
        <a
          class="text-blue-tl text-sm"
          href="#"
        >
          Where can I find this info?
        </a>
      </div>
      <div
        class="flex flex-col py-6 space-y-6 sm:space-y-0 sm:space-x-12 sm:flex-row"
      >
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="account_id"
            >
              Account ID
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="account_id"
                name="account_id"
                type="text"
                value=""
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="public_key"
            >
              Public Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="public_key"
                name="public_key"
                type="text"
                value=""
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="private_key"
            >
              Private Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="private_key"
                name="private_key"
                type="text"
                value=""
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="approved_roles"
            >
              What Roles Provide Support?
            </label>
            <div
              class="mt-2"
            >
              <div
                class="multiselect-container multiSelectContainer "
                id="approved_roles"
              >
                <div
                  class="search-wrapper searchWrapper "
                >
                  <input
                    autocomplete="off"
                    class="searchBox"
                    id="approved_roles_input"
                    placeholder="Select"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="optionListContainer displayNone"
                >
                  <ul
                    class="optionContainer"
                  >
                    <li
                      class="
              highlightOption highlight 
              false 
              false option
            "
                    >
                      Administrator
                    </li>
                    <li
                      class="
               
              false 
              false option
            "
                    >
                      Editor
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="helpdesk"
            >
              Helpdesk
            </label>
            <div
              class="mt-2"
            >
              <select
                class="bg-white block w-full pl-3 pr-8 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="helpdesk"
                name="helpdesk"
              >
                <option>
                  Select a Help Desk
                </option>
                <option
                  value="helpscout"
                >
                  Help Scout
                </option>
                <option
                  value="zendesk"
                >
                  Zendesk
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pt-8 mt-4 border-t"
      >
        <div
          class="flex justify-end"
        >
          <button
            class="bg-white py-2 px-4 border border-gray-300 rounded-lg text-sm font-medium text-gray-900 hover:bg-gray-50 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Cancel
          </button>
          <button
            class="ml-3 inline-flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-lg text-white bg-blue-tl hover:bg-indigo-700 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Add Team
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`EditTeam Renders & Matches snapshot 1`] = `
<div>
  <form
    class="flex px-5 pt-20 sm:px-10"
  >
    <div
      class="flex flex-col w-full max-w-4xl mx-auto p-8 bg-white rounded-lg shadow sm:p-14 sm:pb-8"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <div
        class="max-w-sm mx-auto mb-8 justify-center text-center"
      >
        <h2
          class="mt-4 text-2xl text-gray-900"
        >
          Update Team
        </h2>
        <p
          class="mt-2 mb-4 text-sm text-gray-500"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ornare tortor in nisl fermentum.
        </p>
        <a
          class="text-blue-tl text-sm"
          href="#"
        >
          Where can I find this info?
        </a>
      </div>
      <div
        class="flex flex-col py-6 space-y-6 sm:space-y-0 sm:space-x-12 sm:flex-row"
      >
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="account_id"
            >
              Account ID
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="account_id"
                name="account_id"
                type="text"
                value="1"
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="public_key"
            >
              Public Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="public_key"
                name="public_key"
                type="text"
                value="fsaffff"
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="private_key"
            >
              Private Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="private_key"
                name="private_key"
                type="text"
                value="asda"
              />
              <div
                class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.06004 6.00004C6.21678 5.55449 6.52614 5.17878 6.93334 4.93946C7.34055 4.70015 7.8193 4.61267 8.28483 4.69252C8.75035 4.77236 9.17259 5.01439 9.47676 5.37573C9.78093 5.73706 9.94741 6.19439 9.94671 6.66671C9.94671 8.00004 7.94671 8.66671 7.94671 8.66671M8.00004 11.3334H8.00671M14.6667 8.00004C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00004C1.33337 4.31814 4.31814 1.33337 8.00004 1.33337C11.6819 1.33337 14.6667 4.31814 14.6667 8.00004Z"
                    stroke="#98A2B3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.33333"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="approved_roles"
            >
              What Roles Provide Support?
            </label>
            <div
              class="mt-2"
            >
              <div
                class="multiselect-container multiSelectContainer "
                id="approved_roles"
              >
                <div
                  class="search-wrapper searchWrapper "
                >
                  <input
                    autocomplete="off"
                    class="searchBox"
                    id="approved_roles_input"
                    placeholder="Select"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="optionListContainer displayNone"
                >
                  <ul
                    class="optionContainer"
                  >
                    <li
                      class="
              highlightOption highlight 
              false 
              false option
            "
                    >
                      Administrator
                    </li>
                    <li
                      class="
               
              false 
              false option
            "
                    >
                      Editor
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="helpdesk"
            >
              Helpdesk
            </label>
            <div
              class="mt-2"
            >
              <select
                class="bg-white block w-full pl-3 pr-8 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="helpdesk"
                name="helpdesk"
              >
                <option>
                  Select a Help Desk
                </option>
                <option
                  value="helpscout"
                >
                  Help Scout
                </option>
                <option
                  value="zendesk"
                >
                  Zendesk
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pt-8 mt-4 border-t"
      >
        <div
          class="flex justify-end"
        >
          <button
            class="bg-white py-2 px-4 border border-gray-300 rounded-lg text-sm font-medium text-gray-900 hover:bg-gray-50 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Cancel
          </button>
          <button
            class="ml-3 inline-flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-lg text-white bg-blue-tl hover:bg-indigo-700 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Update Team
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`NoTeams Renders & Matches snapshot 1`] = `
<div>
  <div
    class="flex px-5 pt-12 sm:px-10 sm:pt-32"
  >
    <div
      class="flex flex-col justify-center text-center w-full max-w-lg mx-auto p-8 bg-white rounded-lg shadow sm:p-14"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <h2
        class="mt-4 text-2xl text-gray-900"
      >
        Create your first team
      </h2>
      <button
        class="inline-flex items-center justify-center px-4 py-2 border border-transparent shadow-sm font-medium rounded-lg text-white bg-blue-tl focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500 sm:text-sm"
        type="button"
      >
        <svg
          class="mr-1"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
          >
            <path
              d="M12 6v6m0 0v6m0-6h6m-6 0H6"
              stroke="#FFFFFF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
        </svg>
        Add Team
      </button>
    </div>
  </div>
</div>
`;
